src/main/java/frc/robot/Robot.java

```diff
@@ -159,12 +193,117 @@ public void robotPeriodic() {
         });
   }
 
+  public Command drivePath() {
+    return Commands.sequence(
+        swerve
+            .runVelocityFieldRelative(
+                () -> {
+                  System.out.println(AutoAimStates.elapsedAutonomousSeconds);
+                  AutoAimStates.curState =
+                      swerve.getAutoState(AutoAimStates.elapsedAutonomousSeconds);
+                  AutoAimStates.elapsedAutonomousSeconds +=
+                      Timer.getFPGATimestamp()
+                          - AutoAimStates.elapsedAutonomousSeconds
+                          - AutoAimStates.startingAutonomousSeconds;
+                  return new ChassisSpeeds(
+                      AutoAimStates.curState.velocityX,
+                      AutoAimStates.curState.velocityY,
+                      AutoAimStates.curState.angularVelocity);
+                })
+            .until(
+                () ->
+                    AutoAimStates.elapsedAutonomousSeconds
+                        >= Choreo.getTrajectory("amp 4 local sgmt 1").getTotalTime()),
+        Commands.runOnce(
+            () -> {
+              AutoAimStates.startingAutonomousSeconds = Timer.getFPGATimestamp();
+            },
+            swerve));
+  }
+
+  /**
+   * A demo command that goes through all the steps of a shoot while moving algorithm Has print
+   * commands for any unimplemented functionality
+   *
+   * @param speeds
+   * @return A command that takes the robot through an auto aim sequence
+   */
+  public Command teleopAutoAim(Supplier<ChassisSpeeds> speeds) {
+
+    return Commands.sequence(
+            Commands.runOnce(
+                () -> {
```

Comment:
This could be clearer if you gave each of the steps names, either by making them actual methods or by naming variables. ie. `var getShotData = Commands.runOnce(() -> { ... })` and then `Commands.runOnce(getShotData, ...)`.